region_code_filter.py

```diff
@@ -461,26 +528,89 @@ def get_landsat_level1_file_paths(
+                    "PBS queue).")
+@click.option("--walltime",
+              help="Job walltime in `hh:mm:ss` format.")
+@click.option("--email",
+              help="Notification email address.")
+@click.option("--project", default="v10", help="Project code to run under.")
+@click.option("--logdir", type=click.Path(file_okay=False, writable=True),
+              help="The base logging and scripts output directory.")
+@click.option("--pkgdir", type=click.Path(file_okay=False, writable=True),
+              help="The base output packaged directory.")
+@click.option("--env", type=click.Path(exists=True, readable=True),
+              help="Environment script to source.")
+@click.option("--workers", type=click.IntRange(1, 48),
+              help="The number of workers to request per node.")
+@click.option("--nodes", help="The number of nodes to request.")
+@click.option("--memory",
+              help="The memory in GB to request per node.")
+@click.option("--jobfs",
+              help="The jobfs memory in GB to request per node.")
 def main(
-    brdf_shapefile: click.Path,
-    one_deg_dsm_v1_shapefile: click.Path,
-    one_sec_dsm_v1_shapefile: click.Path,
-    one_deg_dsm_v2_shapefile: click.Path,
-    satellite_data_provider: str,
-    aerosol_shapefile: click.Path,
-    world_wrs_shapefile: click.Path,
-    world_mgrs_shapefile: click.Path,
-    usgs_level1_files: click.Path,
-    search_datacube: bool,
-    allowed_codes: click.Path,
-    nprocs: int,
-    config: click.Path,
-    days_delta: int,
-    products: list,
-):
+        brdf_shapefile: click.Path,
+        one_deg_dsm_v1_shapefile: click.Path,
+        one_sec_dsm_v1_shapefile: click.Path,
+        one_deg_dsm_v2_shapefile: click.Path,
+        satellite_data_provider: str,
+        aerosol_shapefile: click.Path,
+        world_wrs_shapefile: click.Path,
+        world_mgrs_shapefile: click.Path,
+        usgs_level1_files: click.Path,
+        search_datacube: bool,
+        allowed_codes: click.Path,
+        nprocs: int,
+        config: click.Path,
+        days_delta: int,
+        products: list,
+        workdir: click.Path,
+        run_ard: bool,
+        landsat_aoi: bool,
+        **ard_click_params: dict,
+    ):
+    """
+    The keys for ard_click_params;
+        test: bool,
+        logdir: click.Path,
+        pkgdir: click.Path,
+        env: click.Path,
+        ardworkers: int,
+        ardnodes: int,
+        ardmemory: int,
+        ardjobfs: int,
+        project: str,
+        walltime: str,
+        email: str
+
+    :return: list of scenes to ARD process
+    """
+    workdir = Path(workdir).resolve()
+    # set up the scene select job dir in the work dir
+    jobid = uuid.uuid4().hex[0:6]
+    jobdir = Path(os.path.join(workdir, FMT2.format(jobid=jobid)))
+
+    #
+    print("Job directory: " + str(jobdir))
+    log_filepath = os.path.join(jobdir, LOG_FILE)
+    if not os.path.exists(jobdir):
```

Comment:
similar to comment above.  If it is a `Path` object, its then probably cleaner to use the `.exists` method.